homeassistant/components/plex/server.py

```diff
@@ -669,6 +669,120 @@ def lookup_media(self, media_type, **kwargs):
+
+        Notes:
+            - Deck mode supports shows, seasons
+            - Unfinished mode supports shows, seasons
+            - Unwatched mode supports shows, seasons
+            - Watched mode supports shows, seasons
+
+        Args:
+            media (plexapi.video.Show or plexapi.video.Season): media to be continued
+            library_name (str): library section used for 'deck' continuation mode
+            continuation (str): structured continuation query as described in this method
+
+        Returns:
+            plexapi.video.Show or plexapi.video.Season object
+
+            Return object will always be either the original media argument, or a more suitable
+            continuation media.
+
+        Examples:
+            Let's assume the following complex continuation mechanism is desired:
+
+                Resume media from deck, and if it's missing then
+                resume last unfinished episode, and if it's missing then
+                resume first unwatched episode, and if it's missing then
+                start show from the beginning
+
+            Resultant continuation query:
+
+                "_continue": "deck|unfinished_last|unwatched_first"
+
+        """
+        if type(media) not in (plexapi.video.Show, plexapi.video.Season):
+            return media
+
+        if not continuation or not media:
+            return media
+
+        for query in continuation.lower().split("|"):
+            # Query format: <mode>[_<parameter>]
+            args = query.split("_")
+            mode = args[0]
+            index = 0
+
+            # Validate mode parameters
+            if len(args) == 2 and args[1] in ("first", "last"):
+                index = 0 if args[1] == "first" else -1
+            elif len(args) != 1:
+                _LOGGER.error("Unrecognized continuation mode args: '%s'", query)
+                continue
+
+            episodes = []
+            if mode == "deck":
+                try:
+                    library_section = self.library.section(library_name)
+                    deck = library_section.onDeck()
+                except NotFound:
+                    _LOGGER.error("Library '%s' not found", library_name)
+                    return media
+
+                if isinstance(library_section, plexapi.library.ShowSection):
+                    if isinstance(media, plexapi.video.Show):
+                        show_name = media.title
+                        season_number = None
+                    elif isinstance(media, plexapi.video.Season):
+                        show_name = media.parentTitle
+                        season_number = media.seasonNumber
+
+                    episodes = [e for e in deck if e.grandparentTitle == show_name]
+                    if season_number is not None:
+                        episodes = [
+                            e for e in episodes if e.seasonNumber == season_number
+                        ]
```

Comment:
Can you instead use Season.episodes() and Show.episodes()?

Reply:
In this case I was searching the deck collection for an appropriate instance of a show/season/episode, and only wanted to gather episodes that were on the deck (rather than Season.episodes() or Show.episodes() which would gather all episodes). May seem silly, but I did this because I couldn't find any assurances in the API docs that deck would not contain multiple episodes from the same show :)

Anyway, since you brought my attention to it - I do think this was needlessly complex, and I found a much cleaner way to do this in a single Plex API onDeck method call:

https://python-plexapi.readthedocs.io/en/latest/modules/video.html#plexapi.video.Show.onDeck
https://python-plexapi.readthedocs.io/en/latest/modules/video.html#plexapi.video.Season.onDeck

New code looks super neat, IMO.